src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -451,6 +451,62 @@ def _is_range_full_extent(my_range):
     return is_lower and is_upper and is_step
 
 
+def _create_array_bound_arg(node):
```

Comment:
It looks like the name explains why is it called by the current caller but not what it does. Could this function be named `_copy_full_base_reference`?

(Another possibility that I am not going to ask in this PR is for this to be methods in Member and Reference, there is nothing frontend specific and this could be useful in other places)

Reply:
Good point, renamed.